### DISCUSSION TITLE
Add custom information to an assertion error message.

### DISCUSSION THREAD

**Discussion by evant** [assertk-org/assertk#352] [Ideas]
This came up as a response to https://github.com/willowtreeapps/assertk/issues/12. It would be useful to add additional information to a failure to aid with tracking down the cause. Some bikeshed syntax (may change):

```kotlin
assertThat(success, context = { "DB state: " + repository.findUser(USER_ID) }).hasSize(1)

// expected [size]:<1> but was:<0>
//   DB state: User { name = "Tim" }"
```

**Comment by neeme-praks-sympower** [comment]
What's the status with this? It does not seem to be implemented in `0.25`?

**Reply by evant** [reply]
Nothing yet, needs some design work. If you have usecases you'd like to share ot would be helpful.

**Reply by neeme-praks-sympower** [reply]
My usecase: the assertion is too specific, I would like to add some context to it. Sample code:
```kotlin
  private fun assertThatEventsStored(expectedEvents: List<Event>) {
    argumentCaptor<Event> {
      verify(eventRepository, times(expectedEvents.size)).save(capture())
      allValues.zip(expectedEvents) { actualEvent, expectedEvent ->
        assertThat(actualEvent.eventType).isEqualTo(expectedEvent.eventType)
        assertThat(actualEvent.equipmentNumber).isEqualTo(expectedEvent.equipmentNumber)
        if (expectedEvent.payload != null) {
          assertThat(actualEvent.payload).isEqualTo(expectedEvent.payload)
        }
      }

    }
    verifyNoMoreInteractions(eventRepository)
  }
```
Now, if `equipmentNumber` or `payload` assertion fails, I would also like to know the associated `eventType` (as it is a loop, line number from the exception does not help me). So I would like to add a message to the assertion failure message (the original failure message would still be there).
